osu.Game/Migrations/20200712012841_KeyBindingActionValueChange.cs

```diff
@@ -0,0 +1,125 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System;
+using System.Collections.Generic;
+using Microsoft.EntityFrameworkCore.Migrations;
+using osu.Framework.Platform;
+using osu.Game.Database;
+using osu.Game.Input.Bindings;
+using osu.Game.Rulesets;
+using Microsoft.EntityFrameworkCore.Migrations;
+
+namespace osu.Game.Migrations
+{
+    public partial class KeyBindingActionValueChange : Migration
+    {
+        private readonly RulesetStore rulesetStore;
+
+        // OR
+
+        private readonly KeybindActionLookup actionLookup;
+
+        public KeyBindingActionValueChange(DatabaseContextFactory databaseContextFactory, Storage storage)
```

Comment:
as you stated, migrations expect a parameterless ctor, so option 1 is not even an option unless the dependencies are manually created.